### PR TITLE
DOCSP-2984: Improve analytics

### PR DESCRIPTION
**Pull request by i80and**

### CODE REVIEW

File: themes/mongodb/src/js/componentAccordion.js [hunk 1]
```diff
@@ -11,6 +13,12 @@ function accordionShowHide(element) {
 
     const control = element.querySelector('.accordion__action');
     control.innerHTML = (control.innerHTML === 'Expand') ? 'Collapse' : 'Expand';
+
+    const title = element.querySelector('.accordion__title').innerText;
+    reportAnalytics('Accordion Toggled', {
+        'title': title,
+        'status': 'expanded'
```
Comment (kirbyk) [inline themes/mongodb/src/js/componentAccordion.js]: Wouldn't this always have a `status` of `expanded`? This should change based on the state of the accordion.

File: themes/mongodb/src/js/componentThirdParty.js [hunk 2]
```diff
@@ -20,8 +38,39 @@ export function initialize() {
...
-export function setup() {
-    window.delighted.survey();
+export function setup(fastNav) {
+    const project = document.body.getAttribute('data-project');
+    try {
+        const sampleFactor = SAMPLE_FACTORS[project] || 0.1;
+        window.delighted.survey({
+            minTimeOnPage: 180,
+            sampleFactor: sampleFactor
+        });
```
Comment (kirbyk) [inline themes/mongodb/src/js/componentThirdParty.js]: Let's also send what project and branch we're on. Like this:
```
...
properties: {
  project,
  branch,
}
```

File: themes/mongodb/src/js/componentThirdParty.js [hunk 3]
```diff
@@ -20,8 +38,39 @@ export function initialize() {
...
+    } catch (err) {
+        console.error(err);
+    }
+
+    // Update Google Analytics
+    try {
+        window.ga('set', 'page', location.pathname + location.query);
```
Comment (kirbyk) [inline themes/mongodb/src/js/componentThirdParty.js]: What is `location.query`? I think we just want the pathname based on [these docs](https://developers.google.com/analytics/devguides/collection/analyticsjs/single-page-applications#handling_multiple_urls_for_the_same_resource).

File: themes/mongodb/src/js/componentThirdParty.js [hunk 4]
```diff
@@ -20,8 +38,39 @@ export function initialize() {
...
+        window.ga('set', 'title', document.title);
```
Comment (kirbyk) [inline themes/mongodb/src/js/componentThirdParty.js]: I don't think we need to do this. I don't think we ever do `ga('send', 'pageview')` did you mean to do that instead?

**Comment by i80and** [comment]
Updated!

File: themes/mongodb/src/js/componentThirdParty.js [hunk 5]
```diff
@@ -20,8 +60,49 @@ export function initialize() {
...
+    // Update Google Analytics
+    waitForFunction('ga', (ga) => {
+        window.ga('set', 'page', location.pathname);
+        window.ga('send', 'pageview');
+    });
+
+    // Update Segment
+    try {
+        window.analytics.page({
+          project: project,
```
Comment (kirbyk) [inline themes/mongodb/src/js/componentThirdParty.js]: I'm not sure why, but Segment doesn't seem to be pulling the correct information when this gets called. Most strangely when testing this on staging it says that the events are happening on prod.

From my testing in the JS console it looks like we can override this erroneous behavior like so:
```
window.analytics.page({
  path: location.pathname,
  url: location.href,
  project,
  branch,
});
```

File: themes/mongodb/src/js/componentThirdParty.js [hunk 6]
```diff
@@ -20,8 +60,49 @@ export function initialize() {
...
+            }
+        });
+    } catch (err) {
+        console.error(err);
+    }
+
+    // Update Google Analytics
+    waitForFunction('ga', (ga) => {
+        window.ga('set', 'page', location.pathname);
+        window.ga('send', 'pageview');
```
[the remark this answers is not in the document]
Comment (i80and) [reply themes/mongodb/src/js/componentThirdParty.js]: Please do remove the initial pageView hit; this will get updated on all loads.

**Comment by i80and** [comment]
Committed in ba0c08bbb53143ed21d6a19a2d71d5234ddc812e

**Pull request closed without merging**